### ISSUE TITLE
len builtin overriden by pylab import

### ISSUE THREAD

**Issue by sibebleuze** [matplotlib/matplotlib#20205]
### Bug report
**Bug summary**
When the entire pylab module is imported into a project, the len builtin function is overriden in some way.
**Code for reproduction**
```python
from pylab import *
x = len("teststring")
print(x)
```
**Actual outcome**
```
10
```
The code does not give a wrong result, x = 10 as expected, but my IDE (PyCharm Professional) gives me a warning saying: 'len' is not callable. This indicates that it somehow differs from the builtin function it is supposed to be, while it has not been redefined in my code.
**Expected outcome**
I expect the len function to be the Python builtin len function, this should not be anything else, even after an import statement. Note that on adding the following code before `x = len("teststring")`, the IDE warning disappears:
```python
if callable(len):
    pass
```
That being said I have no clue what causes any of the above and I wish you good luck in finding out.
**Matplotlib version**
  * Operating system: Windows 10 Home, version 20H2 (12/‎12/‎2019), build 19042.964
  * Matplotlib version (`import matplotlib; print(matplotlib.__version__)`): 3.3.4
  * Matplotlib backend (`print(matplotlib.get_backend())`): module://backend_interagg
  * Python version (`import platform; platform.python_version()`): 3.8.8
  * PyCharm Professional version: 2021.1.1, build #PY-211.7142.13

Python and matplotlib were installed with conda (default channel) from Anaconda 3, version 2020.11.

**Comment by jklymak** [comment]
1. don't use pylab
2. pylab doesn't redefine `len`, so you should open this issue with pycharm.  I'm going to close here because I don't think we can help you.  Thanks for your understanding.

**Comment by tacaswell** [comment]
I suspect that pycharm is reacting to the `from foo import *` line.  As that step can clobber _any_ name in your namespace, all bets are off from a static analysis point of view.   Jody is correct that we do not change `len` from pylab:

```
✔ 16:15:15 $ ipython
Python 3.9.4 (default, Apr 20 2021, 15:51:38) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.19.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: id(len)
Out[1]: 140374024905168

In [2]: from pylab import *

In [3]: id(len)
Out[3]: 140374024905168

In [4]: 
```

That said, please do not use `pylab`, and if you do want to use it, please do not do `from pylab import *`.  We do clobber some other built-ins (like `sum` is replaced by `np.sum`) which can lead to confusing behavior in some edge cases and * imports tend to confuse static analysis tools.

**Comment by sibebleuze** [comment]
Thanks for all the your comments and answers. You suggested I should not have pinned this on pylab, and you were correct. I tested this with a completely different package (`from imageio import *`) and there it didn't happen. It did however reappear when I did `from numpy import *` (which is in pylab, so that's why pylab causes it as well). I will look into this further and try to find out if any of the underlying imports cause it and try to find out who can fix it (or eventually go to PyCharm and have them take a look at it). Thanks again and sorry for the trouble, I should have looked better before coming here.
